### ISSUE TITLE
Exits on Unhandled Exception

### ISSUE THREAD

**Issue by myhndl** [polyswarm/polyswarm-client#144]
**Describe the bug**
We're exiting when an unhandled exception occurs, e.g. a timeout.

It happening on Tachyon:
https://app.datadoghq.com/logs?cols=%5B%22core_host%22%2C%22core_service%22%2C%22log_http.status_code%22%2C%22log_request_Authorization%22%5D&context_event=AAAAAWgN3Mowp4_NmgAAAABBV2dOM01wQ1hvTjBBelpqc1d1Wg&event=AAAAAWgN3Mowp4_NjQAAAABBV2dOM01wQlhvTjBBelpqc1d1Sw&from_ts=1546280108117&index=main&live=false&logset=main&messageDisplay=inline&query=host%3AEC2AMAZ-DALA5CT+service%3Amicroengine+filename%3Aengine.log&saved_view&stream_sort=desc&to_ts=1546452531421

**To Reproduce**
Run a microengine for a while.

**Expected behavior**
We should not exit; polyswarm-client must do everything to remain available.

**Comment by myhndl** [comment]
Also affecting `consumer_backend`:
```
user@user-VirtualBox:~/work/infrastructure$ docker service logs -t --tail 3 pgzbblzj9o2a
2019-01-02T20:20:23.697482206Z consumer_backend.1.pgzbblzj9o2a@ip-172-31-82-214    | {"level": "INFO", "name": "consumer.ambassador", "timestamp": "2019-01-02T20:20:23.697257Z", "message": "Vote received for bounty c645e5a1-d71b-4568-8f2a-4c40d71b0da5 0xd8b48Da78188312c5fC079E532afd48De973767E: [False]"}
2019-01-02T20:20:23.698995015Z consumer_backend.1.pgzbblzj9o2a@ip-172-31-82-214    | {"level": "INFO", "name": "consumer.ambassador", "timestamp": "2019-01-02T20:20:23.698635Z", "message": "Vote on file c03d23b734567a3deba88fc58cf721f5233e9c958f03268d72984a2bf9ac48a5: False"}
2019-01-02T20:20:23.718341227Z consumer_backend.1.pgzbblzj9o2a@ip-172-31-82-214    | {"level": "ERROR", "name": "polyswarmclient", "timestamp": "2019-01-02T20:20:23.718108Z", "message": "Detected unhandled exception, exiting with failure"}
```

**Comment by myhndl** [comment]
Also affecting `ambassador`:
```
user@user-VirtualBox:~/work/infrastructure$ docker service logs -t --tail 4 lef2fult35tt
2019-01-02T23:30:43.421474033Z epoch_ambassador.1.lef2fult35tt@ip-172-31-82-214    | {"level": "INFO", "name": "ambassador.filesystem", "timestamp": "2019-01-02T23:30:43.421226Z", "message": "Submitting file /tmp/tmpgie9g1stmalware-corpus/benign/65fr2439.pdf"}
2019-01-02T23:30:43.454729577Z epoch_ambassador.1.lef2fult35tt@ip-172-31-82-214    | {"level": "ERROR", "name": "polyswarmclient", "timestamp": "2019-01-02T23:30:43.454522Z", "message": "Received unexpected failure response from polyswarmd", "extra": {"status": "FAIL", "errors": "Could not add artifacts to IPFS"}}
2019-01-02T23:30:43.454817345Z epoch_ambassador.1.lef2fult35tt@ip-172-31-82-214    | {"level": "ERROR", "name": "ambassador.filesystem", "timestamp": "2019-01-02T23:30:43.454701Z", "message": "Could not submit artifact to IPFS"}
2019-01-02T23:30:43.454991316Z epoch_ambassador.1.lef2fult35tt@ip-172-31-82-214    | {"level": "ERROR", "name": "polyswarmclient", "timestamp": "2019-01-02T23:30:43.454854Z", "message": "Detected unhandled exception, exiting with failure"}
```

**Comment by mjkoo** [comment]
Closed by #152
